frontend/src/registration/components/InstanceSetupPage/InstanceSetupPage.tsx

```diff
@@ -58,12 +95,25 @@ export class InstanceSetupPage extends React.PureComponent<Props> {
             <FormLabel>
               <WrappedMessage id="publicContactEmail" messages={messages} />
             </FormLabel>
-            <FormControl type="email" />
+            <FormControl
+              type="email"
+              name="publicContactEmail"
+              value={this.props.publicContactEmail}
+              onChange={this.onChange}
+            />
             <p>
               <WrappedMessage id="publicContactEmailHelp" messages={messages} />
             </p>
           </FormGroup>
         </Form>
```

Comment:
We should add some validation to this form - instance name shouldn't be empty and email should match something like `/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)\|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])\|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/`.

Reply:
I think we can just rely on the server side for this. Otherwise we will need to keep the validation in sync.